experimental/test/test_CFL_reachability.c

```diff
@@ -0,0 +1,548 @@
+    LAGraph_CFL_reachability(outputs, adj_matrices, grammar.terms_count,                 \
+                             grammar.nonterms_count, grammar.rules, grammar.rules_count, \
+                             msg)
+
+#define check_error(error)                                                               \
+    {                                                                                    \
+        retval = run_algorithm();                                                        \
+        TEST_CHECK(retval == error);                                                     \
+        TEST_MSG("retval = %d (%s)", retval, msg);                                       \
+    }
+
+#define check_result(result)                                                             \
+    {                                                                                    \
+        char *expected = output_to_str(0);                                               \
+        TEST_CHECK(strcmp(result, expected) == 0);                                       \
+        TEST_MSG("Wrong result. Actual: %s", expected);                                  \
+    }
+
+typedef struct {
+    size_t nonterms_count;
+    size_t terms_count;
+    size_t rules_count;
+    LAGraph_rule_WCNF *rules;
+} grammar_t;
+
+GrB_Matrix *adj_matrices = NULL;
+GrB_Matrix *outputs = NULL;
+grammar_t grammar = {0, 0, 0, NULL};
+char msg[LAGRAPH_MSG_LEN];
+
+void setup() { LAGraph_Init(msg); }
+
+void teardown(void) { LAGraph_Finalize(msg); }
+
+void init_outputs() { outputs = calloc(grammar.nonterms_count, sizeof(GrB_Matrix)); }
+
+char *output_to_str(size_t nonterm) {
+    GrB_Index nnz = 0;
+    OK(GrB_Matrix_nvals(&nnz, outputs[nonterm]));
+    GrB_Index *row = malloc(nnz * sizeof(GrB_Index));
+    GrB_Index *col = malloc(nnz * sizeof(GrB_Index));
+    bool *val = malloc(nnz * sizeof(GrB_Index));
+    OK(GrB_Matrix_extractTuples(row, col, val, &nnz, outputs[nonterm]));
+
+    // 11 - size of " (%ld, %ld)"
+    char *result_str = malloc(11 * nnz * sizeof(char));
+    result_str[0] = '\0';
+    for (size_t i = 0; i < nnz; i++) {
+        sprintf(result_str + strlen(result_str), i == 0 ? "(%ld, %ld)" : " (%ld, %ld)",
+                row[i], col[i]);
+    }
+
+    free(row);
+    free(col);
+    free(val);
+
+    return result_str;
+}
+
+void free_workspace() {
+
+    for (size_t i = 0; i < grammar.terms_count; i++) {
+        if (adj_matrices == NULL)
+            break;
+
+        if (adj_matrices[i] == NULL)
+            continue;
+
+        GrB_free(&adj_matrices[i]);
+    }
+    free(adj_matrices);
+    adj_matrices = NULL;
+
+    for (size_t i = 0; i < grammar.nonterms_count; i++) {
+        if (outputs == NULL)
+            break;
+
+        if (outputs[i] == NULL)
+            continue;
+
+        GrB_free(&outputs[i]);
+    }
+    free(outputs);
+    outputs = NULL;
+
+    free(grammar.rules);
+    grammar = (grammar_t){0, 0, 0, NULL};
+}
+
```

Comment:
Please, add an classical test:

Grammar:

S -> a b
S -> a S b

Graph1 
0 -[a] -> 1
1 -[a] -> 0
0 -[b] -> 2
2 -[b] -> 3
3 -[b] -> 0

Graph2 
0 -[a] -> 1
1 -[a] -> 0
0 -[b] -> 0

Reply:
`graph_double_cycle` is Graph1 in my tests and there is test with this grammar and graph